src/bpm/runc/adapter/adapter.go

```diff
@@ -288,13 +289,23 @@ func (a *RuncAdapter) BuildSpec(
 	return *spec, nil
 }
 
-func filterVolumesUnderBoshMounts(mounts []specs.Mount, volumes []config.Volume) []config.Volume {
+func filterVolumesUnderBoshMounts(boshMounts []specs.Mount, unrestrictedVolumes []config.Volume) []config.Volume {
 	var filteredVolumes []config.Volume
-	for _, v := range volumes {
+	for _, v := range unrestrictedVolumes {
 		keep := true
-		for _, m := range mounts {
-			if strings.HasPrefix(v.Path, m.Destination) {
-				keep = false
+		for _, m := range boshMounts {
+			// Check to see whether the "directory parts" of the volume are a sub-set of and existing BPM-default
+			// directory (that will already be mounted) so that we do not accidentally filter out mounts which
+			// have a name that is a sub-string of the existing job. For example the job `service-metrics` should be
+			// able to have an unrestricted volume mount of the `service-metrics-adapter` job directory.
+			boshMountDirParts := strings.Split(m.Destination, "/")
```

Comment:
Might be nicer to use [filepath](https://pkg.go.dev/path/filepath#SplitList), which has knowledge of the separator used so you don't have to hardcode it to `/`.

Reply:
updated, thanks!